hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializer.java

```diff
@@ -209,7 +215,17 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 		if ( lazyInitializer != null ) {
 			lazyInitializer.setUnwrap( unwrapProxy );
 		}
+
 		isInitialized = true;
+
+		if ( entityInstance == null ) {
+			rowProcessingState.getJdbcValuesSourceProcessingState()
```

Comment:
Maybe move that up into the if that already checks if `entityInstance` is `null`. Also, a code comment that explains why the `LoadingEntityEntry` is registered would be nice. While we are at it, shouldn't we always register a `LoadingEntityEntry` when `existingLoadingEntry` is null?

Reply:
not sure

Reply:
I may be wrong but this seems to me the only case where this is really needed

Reply:
What if we have two associations referring to the same entity type with batching. Then I think we'd try to load the entities twice since we don't register the loading entity entry.

Reply:
if you give me a test and I'll check

Reply:
You're right, I checked the code again and this is fine.